[c10] Adjust macro check that detects if glibc++ use c99 csqrt

malfet opened the pull request:
Summary:
This fixes `warning: '_GLIBCXX11_USE_C99_COMPLEX' is not defined, evaluates to 0`, that would be raised if https://github.com/pytorch/pytorch/pull/54820 used with libstd++ compiled without USE_C99_COMPLEX support.

In `c++config.h` `_GLIBCXX_USE_C99_COMPLEX` is aliased to either `_GLIBCXX99_USE_C99_COMPLEX` or `_GLIBCXX11_USE_C99_COMPLEX` depending on  `__cplusplus` macro, as shown here:
https://github.com/gcc-mirror/gcc/blob/0cf4813202f19768e31666f6aa82bde4dce4065a/libstdc%2B%2B-v3/include/bits/c%2B%2Bconfig#L641-L647

Abovementioned config file is generated by autoconf, that leaves macro undefined if feature is not used, so using conditional like `defined(_GLIBCXX_USE_C99_COMPLEX) && _GLIBCXX_USE_C99_COMPLEX == 0` would trigger undefined macro preprocessor warning.

Test Plan: CI

Differential Revision: D27517788

janeyx99 reviewed (approved):
I trust this if the CI passes. Also, just making sure for myself, you mean _GLIBCXX**98**_USE_C99_COMPLEX here instead of _GLIBCXX99_USE_C99_COMPLEX, right?

malfet committed:
[c10] Adjust macro check that detects if glibc++ use c99 csqrt (#55177)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/55177

This fixes `warning: '_GLIBCXX11_USE_C99_COMPLEX' is not defined, evaluates to 0`, that would be raised if https://github.com/pytorch/pytorch/pull/54820 used with libstd++ compiled without USE_C99_COMPLEX support.

In `c++config.h` `_GLIBCXX_USE_C99_COMPLEX` is aliased to either `_GLIBCXX99_USE_C98_COMPLEX` or `_GLIBCXX11_USE_C99_COMPLEX` depending on  `__cplusplus` macro, as shown here:
https://github.com/gcc-mirror/gcc/blob/0cf4813202f19768e31666f6aa82bde4dce4065a/libstdc%2B%2B-v3/include/bits/c%2B%2Bconfig#L641-L647

Abovementioned config file is generated by autoconf, that leaves macro undefined if feature is not used, so using conditional like `defined(_GLIBCXX_USE_C99_COMPLEX) && _GLIBCXX_USE_C99_COMPLEX == 0` would trigger undefined macro preprocessor warning.

Test Plan: CI

Reviewed By: Orvid

Differential Revision: D27517788

fbshipit-source-id: f6c914908288f37aa801f8202947a2c617b7c523
